share/shwap/p2p/bitswap/getter.go

```diff
@@ -57,12 +58,24 @@ func NewGetter(
 // with regular full node peers.
 func (g *Getter) Start() {
 	ctx, cancel := context.WithCancel(context.Background())
-	g.availableSession = g.exchange.NewSession(ctx)
-	g.archivalSession = g.exchange.NewSession(ctx)
 	g.cancel = cancel
+
+	var availableID atomic.Uint64
+	g.availablePool.New = func() interface{} {
+		log.Debugw("creating new available window session", "id", availableID.Load())
+		availableID.Add(1)
```

Comment:
Loaded value might be overwriten between add, so logging might be confusing. Would suggest rather to call  atomic.Add() first and use value returned by Add() for logging. 
```suggestion
	g.availablePool.New = func() interface{} {
	        id := availableID.Add(1)
		log.Debugw("creating new available window session", "id", id)
		
```